PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,332 @@
+    Returns: classes
+    Return type: list
+    """
+    clss = driver.find_elements_by_class_name("myapps-item")
+    t = type(clss)
+    print(t)
+    if not clss:
+        logging.info("No app list found！")
+        return
+    classes = []
+    for cls in clss:
+        txt = cls.find_element_by_class_name("myapps-item-label").text
+        txt = str.strip(txt)
+        if ("Look Up Classes".lower() == txt.lower()):
+            classes = cls
+            break
+    if not classes:
+        logging.info("No Look Up Classes feature found in the app list!")
+        return
+    return classes
+
+
+# Go to the advanced options page and start filling in various search terms
+def fillAdvanceSearch(driver):
+    """
+    Fill advance search.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    # Select all options in Subject list
+    subjectList = driver.find_element_by_id("subj_id")  # web element
+    subjectOptions = subjectList.find_elements_by_tag_name("option")  # list
+    subjectListSelect = Select(subjectList)
+    logging.info("Start to select all the contents in the multi-selection drop-down box. There are many options. You may have to wait for more time. Don't worry ...")
+    for i in range(0, len(subjectOptions)):
+        subjectListSelect.select_by_index(i)
+    # Submit search and click
+    sectionSearch(driver)
+
+
+def sectionSearch(driver):
+    """
+    Locate section search button and click.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    ipts = driver.find_elements_by_tag_name("input")  # list of Web Element
+    submit = ""  # webElement
+    for ipt in ipts:
+        if "submit" == ipt.get_attribute("type").lower() and \
+                "Section Search".lower() == ipt.get_attribute("value").lower():
+            submit = ipt
+            break
+    if not submit:
+        logging.info("No submit button found! ! ")
+        sys.exit(-1)  # Program exit
+    #Click the submit button
+    submit.click()
+    # Wait for a while, wait for the page to open and make sure it loads
+    time.sleep(3)
+
+
+# Save the results of the course
+def saveResult(driver):
+    """
+    Save result to a html.
+
+    Parameters: webdriver
+    Returns: html
+    Return type: String
+    """
+    waitUtilPageLoaded(driver)
+    # Scroll down a page to show the table (no practical meaning, just look at the effect)
+    actions = ActionChains(driver)
+    actions.send_keys(Keys.PAGE_DOWN).perform()
+    # After loading is complete, the entire form page
+    html = driver.page_source
+    time.sleep(5)  # Wait for a while, let people take a look, in fact, it does not make sense to the programmer
+    return html
+
+
+# Wait until the page is loaded! This only applies to child pages, not to the main page, because there is no banner_copyright information in the main page
+def waitUtilPageLoaded(driver):
+    """
+    Wait until page loaded.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    while True:
```

Comment:
Not sure if it will cause an infinite loop if the banner copyright always can not be found.